tests/expected/cpp/30073-lineEndings-Unix.cpp

```diff
@@ -0,0 +1 @@
+int main ()
{
   a    = 5;
   bbbb = 6.0;
   int a      = 5;
   float bbbb = 6.0;

   bbbb = 1.0
}

```

Comment:
Interesting, Github now (selectively) is showing line ends? 
`0xD` = `13` = `\r` = `CR` = `␍`  ... they could use any one of those but no they'd rather use: 13th letter ->`^M`? 

Well enough ranting. Why is a \r used here instead of \n?

Reply:
...because the test is 'lineEndings-to-Mac.cfg'?

(`^M`, i.e. control-M, is how one types 0x0D in a terminal. And how vi[m] — and I suspect, git — displays a 0x0D in a file with mixed line endings. You can try it yourself; type `echo -e '`, then control-V, control-M, then `' | od -x`... Note that your command line will *look* like `echo -n '^M' | od -c`, but the `^M` will actually be one character.)

Reply:
Ahh ok, everything works as intended then.

Nice example, I knew that it had to do something with the key sequence ctrl+m, so far I never had to actually use that. In my opinion that feels somewhat arcane and it would be more helpful to use any of the other possibilities. As you mentioned they most likely just grab that from git.